bms_checkstate.c

```diff
@@ -0,0 +1,103 @@
+*Func desc : This function check if the passed battery parameter value is within the specified range 
+*Param     : param_value - The measured battery parameter value, whole limits are to be verified - float type
+*Return    : Returns the status of param limit check
+			 0 - Param is within range (OK)
+			 1 - Param is less than required charge (NOT_OK)
+			 2 - Param is more than desired charge value (NOT_OK)
+*****************************************************************************************/	
+int BatteryParamCheck(float param_value, float min_value, float max_value)
+{
+    if(param_value < min_value)
+	{
+		return 1;
+	}
+	else if (param_value > max_value) 
+	{
+		return 2;
+	}
+	else
+	{
+		return 0;
+	}
+}	
+
+/****************************************************************************************
+*Func desc : This function check if the battery state is ok. In this function all the desired battery
+			 parameters are checked to validate the current battery status. Further 
+*Param     : temperature - The current battery temperature value that was measured -float type
+*			 soc         - The current battery state of charge that was measured   -float type	
+*            chargerate  - The charging rate that is measured - float type
+*Return    : Returns the state of battery as int, 
+			 1 - battery is ok
+			 0 - battery is not ok
+*****************************************************************************************/	  
+int BatteryStateCheck(float temperature, float soc, float chargeRate, struct batteryParam_unit unit) 
+{
+	/*All check param functions must return 0 if the param check is OK*/
+    int batteryStatus;
+	float convTemp = convertToCelcius (temperature , unit.tempUnit);
+	/*All check param functions must return 0 if the param check is OK*/
+	int temp_status  = BatteryParamCheck(convTemp , MINTEMP , MAXTEMP);
+	int soc_status = BatteryParamCheck(soc , MINSOC , MAXSOC);
+	int chargeRate_status  = BatteryParamCheck(chargeRate , MINCHGRATE , MAXCHGRATE);
+	
+	batteryStatus = !(temp_status || soc_status || chargeRate_status);
+	
+	printBatteryStatus(batteryStatus);
+	
+    return batteryStatus;
+}
+
+/****************************************************************************************
+*Func desc : This function check if the battery state is ok. In this function all the desired battery
+			 parameters are checked to validate the current battery status. 
+*Param     : temperature - The current battery temperature value that was measured -float type
+*			 soc         - The current battery state of charge that was measured   -float type	
+*            chargerate  - The charging rate that is measured - float type
+*Return    : Returns the state of battery as int, 
+			 1 - battery is ok
+			 0 - battery is not ok
+*****************************************************************************************/	
+void printBatteryStatus(int batteryStatus)
+{
+	
+	/*Array 0- English, Array 1- German */
```

Comment:
Nice way to keep the conversion to language till the end and isolate the printing from the rest of the story!
One disadvantage is that all languages get delivered in the binary we build. Any way to avoid that?

Reply:
Due to complexity restriction and introduction of multipe func call i preferred this approach, for the ease of adding more language i used array. Let me know if any simpler approaches are possible